### PR TITLE
[Loading]: .close() can be called earlier than visible=true

### PR DESCRIPTION
**Pull request by koresar**
TL;DR: Sometimes the async sequence of operations can be: `this.visible = false; this.visible = true;` leaving the "Loading" opened forever.

In some scenarios the `.close()` can be called before
```
  Vue.nextTick(() => {
    instance.visible = true;
  });
```
Scenarios include some in-memory caching (Relay/Apollo GraphQL for example).

First time the "Loading" div shows fine. The second time (when a value is cached in memory) it shows and stays forever because `.close()` was called right after the `Loading` constructor call.

The proposed fix solves the bug. Not sure if that's the right way, or if I broke anything else. Probably need testing from core maintainers.

Happy to rework if needed.

### CODE REVIEW

**Comment by wacky6** [comment]
Recommend add tests for your use case.
Also, have a look at the failed test.
